src/main/java/com/aws/greengrass/util/RetryUtils.java

```diff
@@ -92,17 +111,25 @@ public static <T> T runWithRetry(DifferentiatedRetryConfig differentiatedRetryCo
                         attemptMap.put(retryConfig, attempt + 1);
 
                         // log the message
-                        LogEventBuilder logBuilder = attempt == 1 || attempt % LOG_ON_FAILURE_COUNT == 0
-                                ? logger.atInfo(taskDescription)
-                                : logger.atDebug(taskDescription);
-                        logBuilder.kv("task-attempt", attempt).setCause(e).log("task failed and will be retried");
+                        LogEventBuilder logBuilder =
+                                attempt == 1 || attempt % LOG_ON_FAILURE_COUNT == 0
+                                        ? logger.atInfo(taskDescription)
+                                        : logger.atDebug(taskDescription);
+                        logBuilder
+                                .kv("task-attempt", attempt)
+                                .setCause(e)
+                                .log("task failed and will be retried");
 
                         // sleep with back-off
                         if (retryInterval == 0) {
                             retryInterval = retryConfig.getInitialRetryInterval().toMillis();
                         }
-                        Thread.sleep(retryInterval / 2 + RANDOM.nextInt((int) (retryInterval / 2 + 1)));
-                        retryInterval = Math.min(retryInterval * 2, retryConfig.getMaxRetryInterval().toMillis());
+                        Thread.sleep(
+                                retryInterval / 2 + RANDOM.nextInt((int) (retryInterval / 2 + 1)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Manual polling with Thread.sleep() is used instead of AWS SDK waiters. This approach is inefficient, error-prone, and can lead to resource wastage. Use AWS SDK waiters (e.g., AmazonEC2Waiters, AmazonS3Waiters) which provide built-in polling mechanisms with exponential backoff. Learn more https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/waiters.html.